test/Errors.test.js

```diff
@@ -2,357 +2,569 @@
 
```

Comment:
so, my reason for rewriting these tests:

- while debugging this, the older tests were a bit unrevealing & cryptic
- improvment on previous `getErrors`/`getErrorsPromise` approach
- using inline snapshots because i think it really makes sense for these - and will reveal things that previous tests did not.. such as when more warnings/errors are emitted than expected.
- we should probably add some tests covering fatal errors (aka thrown) - any advice welcome